api/src/components/reference/__tests__/updateAllReferences.test.ts

```diff
@@ -0,0 +1,76 @@
+import * as testUtils from 'lib/testUtils';
+
+describe('update all references', () => {
+    beforeEach(async () => {
+        await testUtils.clearDB();
+        await testUtils.testSeed();
+    });
+
+    test('User can update all references from their own draft publication', async () => {
+        const reference = await testUtils.agent
+            .put('/publications/publication-interpretation-draft/reference')
+            .query({ apiKey: '123456789' })
+            .send([
+                {
+                    id: '04',
+                    location: 'http://octopus.ac',
+                    publicationId: 'publication-problem-draft',
+                    text: '<p>Reference 1</p>',
+                    type: 'TEXT'
+                }
+            ]);
+
+        expect(reference.status).toEqual(200);
+    });
+
+    test('User must be the author of the publication to update the references', async () => {
+        const reference = await testUtils.agent
+            .put('/publications/publication-interpretation-draft/reference')
+            .query({ apiKey: '987654321' })
+            .send([
+                {
+                    id: '04',
+                    location: 'http://octopus.ac',
+                    publicationId: 'publication-problem-draft',
+                    text: '<p>Reference 1</p>',
+                    type: 'TEXT'
+                }
+            ]);
+
+        expect(reference.status).toEqual(403);
+    });
+
+    test('The author can only update the references for a live publication', async () => {
+        const reference = await testUtils.agent
+            .put('/publications/publication-real-world-application-live/reference')
+            .query({ apiKey: '123456789' })
+            .send([
+                {
+                    id: '04',
+                    location: 'http://octopus.ac',
+                    publicationId: 'publication-real-world-application-live',
+                    text: '<p>Reference 1</p>',
+                    type: 'TEXT'
+                }
+            ]);
+
+        expect(reference.status).toEqual(403);
+    });
+
+    test('All of the references should be replaced with the new references ', async () => {
+        const reference = await testUtils.agent
+            .put('/publications/publication-interpretation-draft/reference')
+            .query({ apiKey: '123456789' })
+            .send([
+                {
+                    id: '04',
+                    location: 'http://octopus.ac',
+                    publicationId: 'publication-problem-draft',
+                    text: '<p>Reference 1</p>',
+                    type: 'TEXT'
+                }
+            ]);
+
+        expect(reference.body.count).toEqual(1);
```

Comment:
I'd expect some checks of the returned data here to show it matches what was sent

Reply:
actually, how is this different to the first test?

Reply:
ok, to expand, each test starts with the same set of data so if the publication already has some references the first test and this test do the same (except one checks the status and one the count of results)

what you might want to do here is two consecutive update calls and check the data in the response changes

you might also want to update more than one reference in some places..

Reply:
ok I've added this in the test 👍